external/dhcpc/dhcpc.c

```diff
@@ -332,7 +338,16 @@ static uint8_t dhcpc_parseoptions(struct dhcpc_state *presult, uint8_t *optptr,
 		case DHCP_OPTION_DNS_SERVER:
 			/* Get the DNS server address in network order */
 
-			memcpy(&presult->dnsaddr.s_addr, optptr + 2, 4);
+			total_dns = (*(optptr + 1)) / 4;                /* Get the total number of DNS server addresses */
+			total_dns = (total_dns > 2) ? 2 : total_dns;
+			uint8_t *start_addr = optptr + 2;
+
+			for (int i = 0; i < total_dns; ++i) {
```

Comment:
Iterator "i" for the 'for' loop should be defined at the start of the function with other variables.